include/boost/thread/pthread/condition_variable_fwd.hpp

```diff
@@ -58,35 +58,31 @@ namespace boost
             // above) and must be initialized (etc) in case some
             // compilation units provide interruptions and others
             // don't.
-            res=pthread_mutex_init(&internal_mutex,NULL);
+            res=posix::pthread_mutex_init(&internal_mutex);
             if(res)
             {
                 boost::throw_exception(thread_resource_error(res, "boost::condition_variable::condition_variable() constructor failed in pthread_mutex_init"));
             }
 //#endif
-            res = pthread::cond_init(cond);
+            res = posix::pthread_cond_init(&cond);
             if (res)
             {
 //#if defined BOOST_THREAD_PROVIDES_INTERRUPTIONS
                 // ditto
-                BOOST_VERIFY(!pthread_mutex_destroy(&internal_mutex));
+                BOOST_VERIFY(!posix::pthread_mutex_destroy(&internal_mutex));
 //#endif
-                boost::throw_exception(thread_resource_error(res, "boost::condition_variable::condition_variable() constructor failed in pthread::cond_init"));
+                boost::throw_exception(thread_resource_error(res, "boost::condition_variable::condition_variable() constructor failed in pthread_cond_init"));
             }
         }
         ~condition_variable()
         {
             int ret;
 //#if defined BOOST_THREAD_PROVIDES_INTERRUPTIONS
             // ditto
-            do {
-              ret = pthread_mutex_destroy(&internal_mutex);
-            } while (ret == EINTR);
+            ret = posix::pthread_mutex_destroy(&internal_mutex);
             BOOST_ASSERT(!ret);
 //#endif
-            do {
-              ret = pthread_cond_destroy(&cond);
-            } while (ret == EINTR);
+            ret = posix::pthread_cond_destroy(&cond);
             BOOST_ASSERT(!ret);
```

Comment:
Please change both `BOOST_ASSERT` into `BOOST_VERIFY` if everything is ok with the logic itself.

```
gcc.compile.c++ ../../../../bin.v2/libs/spirit/classic/test/grammar_mt_tests.test/gcc-8/release/cxxstd-03-iso/threadapi-pthread/threading-multi/visibility-hidden/grammar_mt_tests.o
In file included from ../../../../boost/thread/pthread/thread_data.hpp:14,
                 from ../../../../boost/thread/thread_only.hpp:17,
                 from ../../../../boost/thread/thread.hpp:12,
                 from grammar_mt_tests.cpp:33:
../../../../boost/thread/pthread/condition_variable_fwd.hpp: In destructor ‘boost::condition_variable::~condition_variable()’:
../../../../boost/thread/pthread/condition_variable_fwd.hpp:79:17: error: variable ‘ret’ set but not used [-Werror=unused-but-set-variable]
             int ret;
                 ^~~
cc1plus: all warnings being treated as errors
    "ccache" "g++-8"   -std=c++03 -fvisibility-inlines-hidden -fPIC -m64 -pthread -O3 -finline-functions -Wno-inline -Wall -Wextra -Werror -fvisibility=hidden -Wno-maybe-uninitialized -Wno-sign-compare -Wno-implicit-fallthrough  -DBOOST_ALL_NO_LIB -DBOOST_ALL_NO_LIB=1 -DBOOST_CHRONO_DYN_LINK=1 -DBOOST_THREAD_BUILD_DLL=1 -DBOOST_THREAD_POSIX -DBOOST_THREAD_USE_DLL=1 -DNDEBUG  -I"." -I"../../../.." -c -o "../../../../bin.v2/libs/spirit/classic/test/grammar_mt_tests.test/gcc-8/release/cxxstd-03-iso/threadapi-pthread/threading-multi/visibility-hidden/grammar_mt_tests.o" "grammar_mt_tests.cpp"
```

Reply:
Ok, I'll submit another PR for this.

Reply:
#278